docs/Using Fleet/GitOps.md

```diff
@@ -569,16 +569,44 @@ Can only be configured for all teams (`org_settings`).
 
 #### mdm
 
-The `mdm` section lets you enable MDM features in Fleet.
+##### apple_business_manager
 
-- `apple_bm_default_team` - is name of the team that macOS hosts in Apple Business Manager automatically enroll to when they're first set up. If empty, hosts will enroll to "No team" (default: `""`).
+- `organization_name` - is the organziation name associated with the Apple Business Manager account.
+- `macos_team` - macOS hosts are automatically added to this team in Fleet when they appear in Apple Business Manager.
+- `ios_team` - iOS hosts are automatically added to this team in Fleet when they appear in Apple Business Manager.
+- `ipados_team` - iPadOS hosts are automatically added to this team in Fleet when they appear in Apple Business Manager.
 
 ##### Example
 
 ```yaml
 org_settings:
   mdm:
-    apple_bm_default_team: "Workstations" # Available in Fleet Premium
+    apple_business_manager: # Available in Fleet Premium
+    - organization_name: Fleet Device Management Inc.
+      macos_team: "💻 Workstations" 
+      ios_team: "📱🏢 Company-owned iPhones"
+      ipados_team: "🔳🏢 Company-owned iPads"
+```
+
+Can only be configured for all teams (`org_settings`).
+
+##### volume_purchasing_program
+
+- `location` - is the name of the location in the Apple Business Manager account.
+- `teams` - is a list of team names. If you choose specific teams, App Store apps in this VPP account will only be available to install on hosts in these teams. If not specified, App Store apps are available to install on hosts in all teams.
+
+##### Example
+
+```yaml
+org_settings:
+  mdm:
+    volume_purchasing_program: # Available in Fleet Premium
+    - location: Fleet Device Management Inc.
+      teams: 
+      - "💻 Workstations" 
```

Comment:
A problem that currently exists with `apple_bm_default_team` is that this API currently uses string identifiers. This means that each team is identified by a specific string value (e.g., "Team_A").

If we change the team's name (e.g., from "Team_A" to "💻 Team_A"), this API will point to a team (e.g., "Team_A") that no longer exists, resulting in errors.

To prevent this issue, we can:
Use Unique Identifiers: Instead of using string names, we can use unique team IDs that do not change even if the team name changes.
Update API References: Ensure that any change in team names updates this endpoint (doesn't sound great at all)
Implement Validation: Add validation checks to ensure the API points to existing teams.
Warn users who are modifying the team name that MDM is using the original team name as an identifier
Move the MDM VPP etc information onto the team API itself instead of in the config

I really think BE should think through how this API should work to avoid using modifiable strings as identifiers.

Reply:
Thanks Rachel!

Reply:
> use unique team IDs that do not change even if the team name changes.

Hey @RachelElysia, @georgekarrv, @roperzh, and I chatted during design review today and the plan is to go w/ the above approach.

If a GitOps user changes a team's name via GitOps but forgets to update it in `volume_purchasing_program.teams` they'll see an error (error message TODO).

Reply:
@noahtalerman What happens if a regular user changes a team name, after VPP is attached to the last team name?

Reply:
Hey @RachelElysia! What do you mean by "regular user" here? I'm not sure I'm getting the workflow you're thinking of...